### PR TITLE
PGDB Refactor & PG Rollback Functionality & Rollback Testing Endpoint

### PR DESCRIPTION
**Pull request by dbkbali**
- Requesting this merge now so that it can be used as the basis for testnet docker images - using refactored code and working rollback functionality

- This request contains fixes for issues https://github.com/ShyftNetwork/go-empyrean/issues/95 https://github.com/ShyftNetwork/go-empyrean/issues/83 https://github.com/ShyftNetwork/go-empyrean/issues/77

- A temporary http endpoint has been provided to test forced rollbacks of the blockchain and pg database on the testnet - this should be removed eventually once the shyft geth messaging system has been architected and implemented.

- Further work is currently being completed to address https://github.com/ShyftNetwork/go-empyrean/issues/78 and for ensuring that balances in the pg database above UINT64 can be handled within the pg database.

### CODE REVIEW

**Review by dutterbutter — APPROVED** [review]
I pulled down and ran geth, it successfully recorded blocks, transactions, internal transactions and updated accounts accordingly.

File: core/genesis.go [hunk 1]
```diff
@@ -150,25 +156,25 @@ func WriteShyftGen(gen *Genesis, block *types.Block) {
...
+			cost = "0"
 			gasPrice = 0
 			//Appending GENESIS to address stored as txHash and From Addr
 			Genesis := []string{"GENESIS_", k.String()}
 			GENESIS := "GENESIS"
 			txHash := strings.Join(Genesis, k.String())
 			//Create the accountNonce, set to 1 (1 incoming tx), format type
 			accountNonce := v.Nonce + 1
-			accountNoncee := strconv.FormatUint(accountNonce, 10)
+			// accountNoncee := strconv.FormatUint(accountNonce, 10)
```
Comment (tim-br) [inline core/genesis.go:171]: This should be removed if not used

File: core/sTypes/stypes.go [hunk 2]
```diff
@@ -24,17 +24,18 @@ type SBlock struct {
...
-	To      string
-	Value   string
-	Gas     uint64
-	GasUsed uint64
-	Input   string
-	Output  string
-	Time    string
+	ID      	int
+	Hash    	string
+	BlockHash 	string
```
Comment (tim-br) [inline core/sTypes/stypes.go:29]: only one line is added (that I can see), we should probably revert the other changes

File: light/odr_test.go [hunk 3]
```diff
@@ -239,6 +271,7 @@ func testChainOdr(t *testing.T, protocol int, fn odrTestFn) {
 	)
 	gspec.MustCommit(ldb)
 	// Assemble the test environment
+
```
Comment (tim-br) [inline light/odr_test.go:274]: new line should be removed

**Comment by dutterbutter** [comment]
The blockexplorer UI / api does not seem to pick up one early blocks....can anyone else confirm this....

File: light/odr_test.go (hunk 3 above)
Comment (dbkbali) [reply light/odr_test.go:274]: Thanks @strangepowers I am also reviewing as getting some strange behaviour with respect to creation of Genesis Accounts as well as when running in Docker vs Local. I am going to delay the merge until I have a chance to further investigate this weekend - were you able to get docker running?

**Comment by dbkbali** [comment]
I am closing this for now - as some further testing is required - due to the fact that we are getting some strange behaviours when running in Docker versus locally

**Pull request closed without merging**